### PR TITLE
Multi Account Provider Capability!

### PR DESCRIPTION
**Pull request by amagana3**
This is an implementation of this feature request: https://github.com/Cox-Automotive/terraform-provider-alks/issues/71. The way this works is quite simple, but basically here's an example of it in the works:

**How to use?**
```
provider "alks" {
  url = "https://alks.coxautoinc.com/rest"
}

provider "alks" {
  url     = "https://alks.coxautoinc.com/rest"
  account = "<account No>"
  role    = "<role>"
  alias   = "second"
}

# CREATE IAM ROLE -- Initial Provider
resource "alks_iamrole" "test_role" {
  name                     = "TEST-DELETE"
  type                     = "AWS CodeBuild"
  include_default_policies = false
  enable_alks_access       = true
}

# CREATE IAM ROLE -- Secondary Provider
resource "alks_iamrole" "test_role_nonprod" {
  provider                 = alks.second
  name                     = "TEST-DELETE"
  type                     = "AWS CodeBuild"
  include_default_policies = false
  enable_alks_access       = true
}
```

As long as you specify another provider with the `account`, `role`, and `alias` attributes, then you're able to generate resources for multiple accounts (THAT YOU HAVE ACCESS TO) in one `terraform apply`. This is pretty powerful stuff, and could be even extended to modules. 

**How was this implemented?**
Here's the flow as it is easier than explaining (look at code :))
1. Get credentials like normal for the initial connection - can think of this as the "main" account access.
2. If the account and role fields are set, then let's generate new STS session for the specified account/role. Note: this will verify if you have access to the account, so no tricks please.
3. Once we validate, and STS keys are generated, set the new session for the provider.
4. Done!

### CODE REVIEW

**Comment by amagana3** [comment]
Note to self: Still looking / researching for a way to return back the keys to the user in case they would like to make calls from the specified account.

**Comment by amagana3** [comment]
### Update
Alright this is a bit of a big update, but the latest code adds the following:
- New data source created for generating ALKS keys to a specific account which is based on the provider used.
- New data source is added to the provider definition.
- Updated docs
- Updated `example/alks.tf` - I hope this example / explanation is clear. Tried to be concise yet thorough.

### How does it work?
The data source when called will generate new keys / account information by context. That context is the provider, so if you're using just one provider it'll return keys and account information of that provider. Otherwise if you have multiple providers, you can add an alias to the provider and use `provider: my_provider.alias` field to specify which keys to generate with the data source. The data source can then be used to configure yet another provider such as AWS or ALKS.

File: config.go [hunk 1]
```diff
@@ -247,3 +262,49 @@ func isValidIAM(arn *string, client *alks.Client) bool {
...
+		log.Println("[INFO] ALKS Client configured")
+
+		return client, nil
+	} else {
+
+		// 4. Alright, new credentials needed - swap em out.
+		client.AccountDetails.Account = newAccDetail
+		client.AccountDetails.Role = c.Role
+
+		newCreds, _ := client.CreateIamSession()
```
Comment (webbbarker) [inline config.go:291]: This line can fail, we should check for an error here and return early if we cannot generate keys.
```suggestion
		newCreds, err := client.CreateIamSession()
		if err != nil {
			return nil, err
		}
```

**Review by webbbarker — CHANGES_REQUESTED** [review]
Still running through tests, but I've found at least one potential error.

**Review by webbbarker — APPROVED** [review]
Good work.  👍

**Pull request merged**